src/main/integrations/last-fm/index.ts

```diff
@@ -40,12 +40,30 @@ export default class LastFM implements IIntegration {
     this.lastfmDetails.token = await this.createToken();
     this.saveSettings();
 
-    shell.openExternal(
-      `https://www.last.fm/api/auth/` + `?api_key=${encodeURIComponent(this.lastfmDetails.api_key)}` + `&token=${encodeURIComponent(this.lastfmDetails.token)}`
-    );
+    let authWindow = new BrowserWindow({
+      width: 800,
+      height: 600,
+      show: false,
+      modal: process.platform !== "darwin",
+      titleBarStyle: "hidden"
+    });
+    return new Promise(resolve => {
+      authWindow.loadURL(
+        `https://www.last.fm/api/auth/` +
+          `?api_key=${encodeURIComponent(this.lastfmDetails.api_key)}` +
+          `&token=${encodeURIComponent(this.lastfmDetails.token)}`
+      );
+      authWindow.show();
+
+      authWindow.on("closed", function () {
+        authWindow = null;
+        resolve(true);
+      });
+    });
   }
 
   private async getSession() {
+    await this.authenticateUser();
```

Comment:
This appears to be requiring a user to get a new Token on each opening of YTM Desktop and not using their existing token from a previous session 🤔

The reason for **not** doing it here, but instead doing it in the `json.error` is to know when the token is invalid and then get a new valid token to use for that session.

Reply:
Ok digging into this. These changes might not be necessary, it seems that no matter what that the token in the store is never getting updated, which is the core of my issues since it just keeps trying to use the same token over and over. 

Trying to figure out what's causing this, and then will re-evaluate changes to the flow